features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/NewOverview/Documents.jsx

```diff
@@ -40,8 +41,10 @@ class Documents extends React.Component {
         };
     }
     componentDidMount() {
-        const API = new Api();
-        const docs = API.getDocuments(this.props.api.id);
+        const { api: { apiType, id } } = this.props;
+        const API = apiType === Api.CONSTS.APIProduct ? new APIProduct() : new Api();
+
+        const docs = API.getDocuments(id);
         const { intl } = this.props;
```

Comment:
shall we add this also to the above props de-structure

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/6779/commits/a5b4d09fd25478c04985b9e5f2eea62f56fed44d